constructors/bq/sql/ST_MAKEENVELOPE.sql

```diff
@@ -0,0 +1,16 @@
+-----------------------------------------------------------------------
+--
+-- Copyright (C) 2021 CARTO
+--
+-----------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION `@@BQ_PROJECTID@@.@@BQ_DATASET_CONSTRUCTORS@@.ST_MAKEENVELOPE`
+    (xmin FLOAT64, ymin FLOAT64, xmax FLOAT64, ymax FLOAT64)
+RETURNS GEOGRAPHY
+OPTIONS (description="Creates a rectangular Polygon from the minimum and maximum values for X and Y")
+AS (
+    ST_MAKEPOLYGON(ST_GeogFromText(CONCAT('LINESTRING(', xmin, ' ', ymin, ',', 
```

Comment:
This might be a micro-optimization, but we could put the vertices in counter-clockwise order and then use `ST_MAKEPOLYGONORIENTED` to save the check and reversal that `ST_MAKEPOLYGON` would do.

Reply:
🤔 but then it would be safer not to trust the caller and use `GREATEST(xmin, xmax)`, `LEAST(xmin, xmax)` etc.

Reply:
I am keeping as simple as PostGIS does (I think they do not make any check). But I could add them without any problem

Reply:
On second thought, maybe we should not secong-guess the user. What if xmin= 179, xmax = -181. By the way, we should check what does ST_MAKEPOLYGON does with this kind of input.

Reply:
```sql
ST_MAKEPOLYGON(ST_GeogFromText('LINESTRING(179 10,179 11,-179 11, -179 10)'))
-- 	 POLYGON((-179 10, -179 11, 179 11, 179 10, -179 10))
ST_MAKEPOLYGON(ST_GeogFromText('LINESTRING(-179 10, -179 11, 179 11, 179 10)'))
-- POLYGON((-179 10, -179 11, 179 11, 179 10, -179 10))
ST_MAKEPOLYGONORIENTED([ST_GeogFromText('LINESTRING(179 10,179 11,-179 11, -179 10)')])
-- POLYGON((179 10, 179 11, -179 11, -179 10, 179 10))
ST_MAKEPOLYGONORIENTED([ST_GeogFromText('LINESTRING(-179 10, -179 11, 179 11, 179 10)')])
-- POLYGON((-179 10, -179 11, 179 11, 179 10, -179 10))
```
So, `ST_MAKEPOLYGON` handles the change of date line well, and `ST_MAKEPOLYGONORIENTED` maintains the input order